dev/tools/controllerbuilder/template/controller/controller.go

```diff
@@ -254,17 +254,22 @@ func (a *Adapter) Export(ctx context.Context) (*unstructured.Unstructured, error
 	if mapCtx.Err() != nil {
 		return nil, mapCtx.Err()
 	}
-	// TODO(user): Update other resource reference 
+	// TODO(user): Update other resource references 
 	parent, err := a.id.Parent()
 	if err != nil {
 		return nil, err
 	}
-	obj.Spec.ProjectRef = &refs.ProjectRef{Name: parent.ProjectID}
+	obj.Spec.ProjectRef = &refs.ProjectRef{External: parent.String()}
 	obj.Spec.Location = parent.Location
 	uObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
 	if err != nil {
 		return nil, err
 	}
+
+	u.SetName(a.actual.Id)
+	u.SetGroupVersionKind(krm.{{.Kind}}GVK)
+	u.SetLabels(a.actual.Labels) // TODO(user): Only keep if the proto resource has a Labels field
```

Comment:
I'm not sure about this. For labels, I think we may want to
1. Remove kcc its own labels (as sanity check)
2. if the `.spec` has a label field, assign to that field.
3. Otherwise, assign to the metadata label.

And we have another question: Shall we check if the CRD is `tf2crd` or `dcl2crd` and add the `alpha.cnrm.cloud.google.com/reconciler: direct` annotation? That depends on how the Export is used (for debugging or re-building the KCC object purpose seems like this is the right approach). What if the resource has reconcile gate? I think we may want to consider more about the Export behavior before putting that in the template, because giving a divergent solution is harder to fix than waiting until the customers really needs the `Export`.

Reply:
Maybe we can just skip labels for now, wdyt @yuwenma ? In any case, I think we should not add `alpha.cnrm.cloud.google.com/reconciler: direct` annotation, and instead make the default behavior to enable the direct reconciler, with an optional label to disable it, for specific resources that require care to migrate.